### ISSUE TITLE
nextflow with `singularity=true`  does not bind shared filesystem to container when using `process.executor = 'slurm'`

### ISSUE THREAD

**Issue by mwalzer** [nextflow-io/nextflow#1031]
## Bug report 
nextflow with `singularity=true`  does not bind shared filesystem to container when using `process.executor = 'slurm'`

### Expected behavior and actual behavior
*expected:*
job is executed, singularity is started with the job directory available (including .command.*), results are written (in shared fs) 
*actual:*
job is executed, singularity is started w/o the shared filesystem bind (/mnt/gluster), singularity command script hence is not available, the workflow terminates.

### Steps to reproduce the problem
Slurm cluster with singularity available, excute any workflow with singularity jobs and more than one job to distribute to the cluster. (My assumption is that nextflow with `process.executor = 'slurm'` is supposed to only work from the shared filesystem, I don't see any dedicated file staging options for slurm nodes).

### Program output 
```
  /bin/bash: line 0: cd: /mnt/gluster/nf/work/a7/7bf99e11dd12ab796e8ee812368999: No such file or directory
  /bin/bash: .command.sh: No such file or directory
```
### Environment 
* Nextflow version: N E X T F L O W  ~  version 19.01.0 
* Java version: 
```
openjdk version "1.8.0_191"
OpenJDK Runtime Environment (build 1.8.0_191-8u191-b12-2ubuntu0.16.04.1-b12)
OpenJDK 64-Bit Server VM (build 25.191-b12, mixed mode)
```
* Operating system: [macOS, Linux, etc]
ubuntu linux with slurm and glusterfs

### Additional context
I looked into the jobs `.command.run` file and it looks fine on first glance. I then shelled into the singularity container on a node, and found that the shared file system is indeed not available. A bit of digging found me, that unless the mount point is available in the container beforehand, singularity will not bind that part of the filesystem. [How are external file systems and paths handled in a Singularity Container?
](https://www.sylabs.io/singularity/faq/). 
So I added the singularity bind command to the `.command.run` file and voilà, the job is executed properly via `sbatch .command.run`. (See files attached!). 
[original.command.run.txt](https://github.com/nextflow-io/nextflow/files/2856706/original.command.run.txt)
[working.command.run.txt](https://github.com/nextflow-io/nextflow/files/2856707/working.command.run.txt)
*In conclusion*, a fix would be to add `-B $job-working-directory:$job-working-directory` to the singularity call generation. Again, my assumption is that nextflow slurm singularity is supposed to be run on a shared filesystem basis.

**Comment by mwalzer** [comment]
> @fmorency 16:51 on gitter
@mwalzer add the bind path to your singularity.conf.
you could also use https://www.nextflow.io/docs/latest/process.html?highlight=containeroption#containeroptions to pass the bind path using -B

**Comment by pditommaso** [comment]
Therefore it can be closed.

**Comment by mwalzer** [comment]
@pditommaso I don't know, it might be good to bind the workflow directory by default. 
Also, putting it *into* the workflow is not really nice - you can't readily exchange them like that. Would be better to put it into the profile, imho.
